### DISCUSSION TITLE
Run on_shutdown despite existing background processes

### DISCUSSION THREAD

**Discussion by mturoci** [encode/starlette#1624] [Q&A] [answered]
According to the [docs](https://www.starlette.io/events/), "The shutdown handlers will run once all connections have been closed, and any in-process background tasks have completed.". Is it possible to make starlette run the shutdown handlers despite existing background processes still in place?

**Comment by Kludex** [comment]
I don't think so? You are using uvicorn, right?

**Reply by mturoci** [reply]
Yes, uvicorn.

**Comment by Kludex** [comment, marked as answer]
It's not possible.
